tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -0,0 +1,67 @@
+package machinepools
+
+import (
+	"context"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	AWSKind           = "Amazonec2Config"
+	AWSPoolType       = "rke-machine-config.cattle.io.amazonec2config"
+	AWSResourceConfig = "amazonec2config"
+)
+
+func NewAWSMachineConfig(generatedPoolName, namespace, region string) *unstructured.Unstructured {
+	machineConfig := &unstructured.Unstructured{}
```

Comment:
use the machineConfig object for this instead of unstructured.

Reply:
kept the unstructured, but with the thought of having constructors with the different provider types.